inc/classes/class-offline-form.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Offline form submission class.
+ *
+ * @package rt-pwa-extensions
+ */
+
+namespace RT\PWA\Inc;
+
+use \RT\PWA\Inc\Traits\Singleton;
+
+/**
+ * Class Offline_Form
+ */
+class Offline_Form {
+
+	use Singleton;
+
+	/**
+	 * @var array of error messages.
+	 */
+	private $error_messages;
+
+	/**
+	 * Construct method.
+	 */
+	protected function __construct() {
+
+		$this->error_messages = array(
+			'serverOffline' => __( 'The server appears to be down. Please try again later.', 'pwa-extension' ),
+			'error'         => __( 'Something prevented the page from being rendered. Please try again.', 'pwa-extension' ),
+			'form'          => __( 'Your form will be submitted once you are back online!', 'pwa-extension' ),
```

Comment:
Better to escape these strings as well here because it will localize to JS file so.